### ISSUE TITLE
Run/Debug Terminal window not appearing when forwarding X

### ISSUE THREAD

**Issue by ngjermundshaug** [eranif/codelite#2248]
Built codelite fresh from source 2 days ago on a aarch64 NVidia Jetson TX2. I am accessing this from a Win10 PC - using Mobalite - with X forwarding. This works fine for all other programs - e.g. Chromium, CodeBlocks etc.

But with codelite - the xterm (terminal) window is not forwarded form the Jetson to my PC when I run/debug the program. (Program runs/debugs fine).

Have tried all terminals - none work. When running e.g. xterm manually - it appears fine.

**Comment by eranif** [comment]
Can you try and configure the display environment variable from within CodeLite? (settings->environment variables)

**Comment by ngjermundshaug** [comment]
![image](https://user-images.githubusercontent.com/1677884/55278734-527cbf00-5310-11e9-869c-5bcb4a8d6bc7.png)

I set DISPLAY in EnvironmentVariables 

![image](https://user-images.githubusercontent.com/1677884/55278755-8bb52f00-5310-11e9-9856-be038f2a6c9f.png)

and then restarted codelite and started the program. Still no window.

**Comment by ngjermundshaug** [comment]
Any other ideas on how I can debug this further? Is anybody else able to recreate this by running CodeLite via SSH and X forwarding ?

**Comment by eranif** [comment]
I did not forget about you :)
I am currently writing a poor mans terminal so it can be embedded into CodeLite as any other Window.
Once it's done, you will be able to use it instead of the current system terminals

**Comment by ngjermundshaug** [comment]
Great, looking forward to give it a try 😊

But having a terminal should be possible though. It works fine with **Codeblocks**. I get a separate xterm window every time run/debug a program. Running standard settings there.

**Comment by ngjermundshaug** [comment]
By adding **--disable-factory** it seems to work - at least I get the window appearing and the application is running. Have only tried it manually. Is there anywhere in codelite or settings files I can easily modify this?

`/usr/bin/gnome-terminal --disable-factory  --working-directory=/home/nvidia/Desktop/steer/src/OperatorControl -e '/bin/bash -f "/home/nvidia/.codelite/tmp/nvidia/codelite-exec.sh"'`

https://askubuntu.com/questions/1050614/how-to-forward-x-over-ssh-to-run-gnome-terminal-remotely-18-04

**Comment by ngjermundshaug** [comment]
![giphy](https://user-images.githubusercontent.com/1677884/55495744-dbac3280-563d-11e9-8dc1-738b7928d1d3.gif)

**Comment by eranif** [comment]
:)

I will try and describe how the consoles are handled by CodeLite.
The base class for the console in CodeLite is `clConsoleBase`. There are 2 major types of console:

`clConsoleCMD` and `clConsoleBash`:

```
    clConsoleBase
        |
        |----- clConsoleCMD
        |
        |----- clConsoleBash                                      
                        |
                        |----- clConsoleOSXTerminal
                        |
                        |----- clConsoleGnomeTerminal 
                     
  
```
Rest of the Linux consoles: (`clConsoleKonsole`, `clConsoleLXTerminal`, `clConsoleMateTerminal`. `clConsoleQTerminal`, `clConsoleXfce4Terminal`) are subclass of `clConsoleGnomeTerminal`

The only difference between them and the `Gnome` one is the startup line.

For example, the c-tor of `clConsoleKonsole` looks like this:

```
clConsoleKonsole::clConsoleKonsole()
{
    SetTerminalCommand("/usr/bin/konsole --separate --workdir %WD% -e '%COMMAND%'");
    SetEmptyTerminalCommand("/usr/bin/konsole --separate --workdir %WD%");
}
```

So, basically, all you need to do is open the relevant console class and update the constructor line.
Have a look here:
https://github.com/eranif/codelite/blob/master/CodeLite/clConsoleGnomeTerminal.cpp#L54

**Comment by ngjermundshaug** [comment]
I already added the flag to CodeLite/clConsoleGnomeTerminal.cpp. Works fine now.

**Comment by eranif** [comment]
Do I need to add it to other type of consoles? or only `gnome-terminal` was giving you hard time?

**Comment by eranif** [comment]
If its working for you now, where is the PR ? :D

**Comment by ngjermundshaug** [comment]
Installed and tested all other terminals. They worked. This flag is only needed for gnome-terminal.

PR - https://github.com/eranif/codelite/pull/2254

**Comment by eranif** [comment]
I am going to revert the the change from PR #2254 
For the simple reason that this flag on various distros was removed. The problem is that it does not ignore it silently, it simply fails to run `gnome-terminal`

For example, running this:

```
$/usr/bin/gnome-terminal --disable-factory --working-directory=/tmp -e '/bin/sleep 1000'
Failed to parse arguments: Option "--disable-factory" is no longer supported in this version of gnome-terminal.
```
Fails to start `gnome-terminal`
